src/helpers/Cosmwasm/API.ts

```diff
@@ -47,10 +55,15 @@ export class Cosmwasm {
 
     public static async GetSigningClient(): Promise<SigningCosmWasmClient> {
         const account = Workspace.GetSelectedAccount();
-        let signer = await DirectSecp256k1HdWallet.fromMnemonic(account.mnemonic, {
+        let signer = await WrapWallet.fromMnemonic(global.workspaceChain.signType, account.mnemonic, {
             prefix: global.workspaceChain.addressPrefix,
         });
-        let gasPrice = global.workspaceChain.defaultGasPrice + global.workspaceChain.chainDenom;
+        let gasDenom = global.workspaceChain.chainGasDenom;
+        if(typeof gasDenom === "undefined" || gasDenom === null || gasDenom === ""){
```

Comment:
```suggestion
        if (!gasDenom) {
```
I think we can replace this to be more simplistic